doc/source/ray-core/patterns/ray-get-too-many-objects.rst

```diff
@@ -0,0 +1,30 @@
+Anti-pattern: Fetching too many objects at once with ray.get causes failure
+===========================================================================
+
+**TLDR:** Avoid calling :ref:`ray.get() <ray-get-ref>` on too many objects since this will lead to heap out-of-memory or object store out-of-space. Instead fetch and process one batch at a time.
+
+If you have a large number of tasks that you want to run in parallel, trying to do ``ray.get()`` on all of them at once could lead to failure with heap out-of-memory or object store out-of-space since Ray needs to fetch all the objects to the local node at the same time.
```

Comment:
```suggestion
If you have a large number of tasks that you want to run in parallel, trying to do ``ray.get()`` on all of them at once could lead to failure with heap out-of-memory or object store out-of-space since Ray needs to fetch all the objects to the caller at the same time.
```